File: src/main/java/co/onmind/microhex/application/dto/UpdateRoleRequest.java
package co.onmind.microhex.application.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;

/**
 * DTO for role update requests.
 */
public class UpdateRoleRequest {
    
    @NotBlank(message = "Role name is required")
    @Size(max = 100, message = "Role name cannot exceed 100 characters")
    private String name;
    
    public UpdateRoleRequest() {}
    
    public UpdateRoleRequest(String name) {
        this.name = name;
    }
    
    public String getName() {
        return name;
    }
    
    public void setName(String name) {
        this.name = name;
    }
}

File: src/main/kotlin/co/onmind/hex/application/dto/UpdateRoleRequest.kt
package co.onmind.hex.application.dto

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Size

data class UpdateRoleRequest(
    @field:NotBlank(message = "Role name cannot be blank")
    @field:Size(max = 100, message = "Role name cannot exceed 100 characters")
    val name: String
)